core/rfb.js

```diff
@@ -1203,6 +1223,14 @@
                     now = (new Date()).getTime();
                     Util.Info("First FBU latency: " + (now - this._timing.fbu_rt_start));
                 }
+
+                // Make sure the previous frame is fully rendered first
+                // to avoid building up an excessive queue
+                if (this._display.pending()) {
+                    this._flushing = true;
```

Comment:
Actually no, because we've already put those three bytes in other variables at this point.

But ideally I'd like to put this code higher, but there isn't a rewind function in the socket object AFAICT.